### PR TITLE
Make Active_games_head an SCP_list

### PR DESCRIPTION
**Pull request by MjnMixael**
Converts Active_games_head to an SCP_list to prepare for the SCPUI API.

### CODE REVIEW

**Comment by JohnAFernandez** [comment]
So, I'm going to push these back to 24.2, but I'm going to review them during the release cycle, so that it's still fresh.

**Review by JohnAFernandez — CHANGES_REQUESTED** [review]
So there's two main things.  I think there's a better loop structure that can be used, mostly for clarity, and there's a use of `new` from the original code that, afaict is causing small memory leaks.  AIUI `std::uniquie_pointer` is a good alternative and should handle disposal of the allocated memory.  The list would have to be changed to an `SCP_list<std::unique_pointer<active_game>>` however.

File: code/network/multiutil.cpp [hunk 1]
```diff
@@ -1607,47 +1607,35 @@ active_game *multi_new_active_game( void )
 	if ( new_game == nullptr ) {
```
[the remark this answers is not in the document]
Comment (MjnMixael) [reply code/network/multiutil.cpp]: Per discussion on Discord, I did away with `new` entirely. The new loop iteration method is also in place. I think that fixes up everything you noticed.

**Review by JohnAFernandez — APPROVED** [review]
Agreed

**Pull request merged**